distributed/utils.py

```diff
@@ -383,10 +383,9 @@ def key_split_group(x):
     >>> key_split_group('x-abcdefab')  # ignores hex
     'x-abcdefab'
     """
-    typ = type(x)
-    if typ is tuple:
+    if isinstance(x, tuple):
         return x[0]
-    elif typ is str:
+    elif isinstance(x, six.string_types):
```

Comment:
We're bound by performance here.  This function is called *many* times.  The use of `type(x) is foo` is intentional. 
 This is significantly faster than calling `isinstance`.